13-TeV-examples/uproot_python/HZZAnalysis.ipynb

```diff
@@ -309,14 +330,19 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 5,
+   "execution_count": 6,
    "metadata": {},
    "outputs": [],
    "source": [
-    "def calc_weight(xsec_weight, mcWeight, scaleFactor_PILEUP,\n",
-    "                scaleFactor_ELE, scaleFactor_MUON, \n",
-    "                scaleFactor_LepTRIGGER ):\n",
-    "    return xsec_weight*mcWeight*scaleFactor_PILEUP*scaleFactor_ELE*scaleFactor_MUON*scaleFactor_LepTRIGGER"
+    "def calc_weight(xsec_weight, events):\n",
```

Comment:
change to vectorized calculation - `events` is expected to be an awkward array with the necessary weights as record fields